### DISCUSSION TITLE
Support AppImage portable mode

### DISCUSSION THREAD

**Discussion by ragsaq** [netblue30/firejail#5252] [General]
Hi,

AppImage supports portable mode: https://docs.appimage.org/user-guide/portable-mode.html

When you have:

```
program.AppImage
program.AppImage.home
```

and run `program`, the settings that would go into `%HOME` are stored in `program.AppImage.home`

if I want to run an app without network access (`firejail --net=none --noprofile --appimage program.AppImage`), what do I need to pass to make it work as if I didn't use firejail (minus the network)?

Right now, settings are discarded on each run.

**Comment by rusty-snake** [comment]
I don't think that it is possible to support this for `--appimage` w/o rewriting  firejail's AI code (which would be good but that an other topic). However you can use firejail's `--private` feature to get the same.

```
firejail --net=none [--other-args] --private=$PWD/too.AppImage.home --appimage too.AppImage
```

**Comment by ragsaq** [comment]
I tried that too, but settings aren't saved. The next time it starts, you get the default settings. only `.bashrc` persists.

i used: `firejail --net=none --private=too.AppImage.home --appimage too.AppImage` in the too.AppImage folder.

Isn't there a way to set the $HOME folder and persists whatever files get changed there?

**Reply by rusty-snake** [reply]
> Isn't there a way to set the $HOME folder and persists whatever files get changed there?

That's what `--private=` does.

Can you

1. Try with an absolute path.
2. Join the sandbox to see where the settings got saved.

**Reply by ragsaq** [reply]
It didn't have to do with absolute paths. It turns out some .AppImage files don't work.

I tried with an AppImage of MKVToolNix successfully but it fails with:

https://github.com/horsicq/DIE-engine/releases/download/3.05/Detect_It_Easy-3.05-x86_64.AppImage
